filebeat/input/filestream/internal/input-logfile/store.go

```diff
@@ -280,6 +427,7 @@ func (r *resource) stateSnapshot() state {
 		TTL:     r.internalState.TTL,
 		Updated: r.internalState.Updated,
 		Cursor:  cursor,
+		Meta:    r.cursorMeta,
 	}
```

Comment:
We might want to change this copyNithNewKey:
- pending operations can not update the copy (no reference to the new resource). Therefore the copy must be marked in sync and pending ops should be ignored (set `pendingCursor` to `nil` and `internalInsync` to true). Alternatively we might have to think about introducing some shared update-state that is aware of active copies, but this might be overkill for now.
- The entry is New, but subject to cleanup due to the current value of TTL. Would it make sense to set `Updated` to `time.Now`? I'm on the fence here. Maybe it makes sense to keep the old Update state, as the actual source we do read from did not change. Is `Updated` also updated by pending operations after an ACK? In this case we might want to set `Update: time.Now()` if the current resource has pending operations, but keep the older timestamp if the resource is "inactive"